From abd2afc5f50d841d9ee4def31daef52f22cc0c3a Mon Sep 17 00:00:00 2001
From: Samuel Riedel <sriedel@lowrisc.org>
Date: Thu, 20 Mar 2025 21:56:56 +0000
Subject: [PATCH 8/8] Remove expected integrity errors

This checked for purposeful integrity errors. However, we do not have
integrity features, so this will fail in Sunburst. Instead, the only
provoked errors we expect here are bad-address-errors.

This fixes:
- `rv_dm_bad_sba_tl_access`
- `rv_dm_autoincr_sba_tl_access`
---
 dv/env/rv_dm_scoreboard.sv | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/dv/env/rv_dm_scoreboard.sv b/dv/env/rv_dm_scoreboard.sv
index 91b7a46..c015cbd 100644
--- a/dv/env/rv_dm_scoreboard.sv
+++ b/dv/env/rv_dm_scoreboard.sv
@@ -293,10 +293,8 @@ class rv_dm_scoreboard extends cip_base_scoreboard #(
       `DV_CHECK_EQ(byte_mask, sba_tl_item.a_mask, msg)
     end
 
-    // d_chan intg error is reported as "other" error and takes precedence over transaction error.
-    if (!sba_tl_item.is_d_chan_intg_ok(.throw_error(0))) begin
-      `DV_CHECK_EQ(sba_item.is_err, jtag_rv_debugger_pkg::SbaErrOther)
-    end else if (sba_tl_item.d_error) begin
+    // There is no integrity checks in Sunburst. Tolerate SbaErrBadAddr errors
+    if (sba_tl_item.d_error) begin
       `DV_CHECK_EQ(sba_item.is_err, jtag_rv_debugger_pkg::SbaErrBadAddr)
     end
     `DV_CHECK_EQ(sba_tl_item.a_source, 0, msg)
-- 
2.34.1

